### ISSUE TITLE
Attempting to register on tenant with user that exists on public site fails

### ISSUE THREAD

**Issue by adamthomason** [hyn/multi-tenant#593]
This one has had me puzzled for a couple of weeks now. For reference, I'm using multi-tenant with Laravel Spark and have integrated github.com/jrean/laravel-user-verification also.

Once I have created a tenant, which I can access and use just fine - everything works as expected - I want to create a user on the tenant site whose email already exists on the public site (for my application, tenants are created once a user is registered and subscribed to my application on the public site, which then generates their own tenant site), however it seems as though form validation checks on the public database.

Are there any traits I can use anywhere to ensure that validation happens only on the current site?

Also, as a bonus, this seems to interfere with github.com/jrean/laravel-user-verification, in that validation is also taking place on the public site.

**Comment by luceos** [comment]
You can force validation to happen using the tenant connection (or system) by prefixing any database related validation rules with the tenant connection name, see

https://laravel-tenancy.com/docs/hyn/5.2/validation

Related to the 3rd party package, you can force models from 3rd party packages onto a tenant or system connection using the tenancy.php configuration file, see:

https://github.com/hyn/multi-tenant/blob/a7366833639b67551357b444cdc8c61e3e027a9e/assets/configs/tenancy.php#L270-L282

**Comment by adamthomason** [comment]
Thanks @luceos, I thought you might mention that.

So I think it's probably beneficial for me to create a User model for both tenant and system, in order to use them both within these arrays?

**Comment by luceos** [comment]
Not necessarily, you can also use the (new) TenantAwareConnection, see:

https://github.com/hyn/multi-tenant/blob/5.x/src/Traits/TenantAwareConnection.php

This sets the connection to Tenant if one has been identified.

**Comment by adamthomason** [comment]
I'm already using that, I had updated from the old version of the trait I had contributed to.

That being said, I'm unsure how to make use of the validation for both tenant and system. Take this as an example:

`$this->validate($request, ['token' => 'required']);`

This line of code will be executed by both system and tenant during login, and so I'm unsure how to convert this to use connection based unique validation.

UPDATE:

```
protected function validateRequest(Request $request)
    {
        $validator = Validator::make($request->all(), [
            'email' => 'required|email'
        ]);

        return $validator->passes();
    }
```

This code (3rd party) is also executed on both tenant and system, and required the same change. Validation is one part of Laravel which confuses me!
